awscli/compat.py

```diff
@@ -87,6 +114,8 @@ def compat_open(filename, mode='r', encoding=None):
 
     raw_input = raw_input
 
+    binary_stdin = sys.stdin
```

Comment:
Is there a reason we're not being consistent and having context managers for `BinaryStdin`, simliar to `BinaryStdout`?  Also, is the `msvcrt.setmode` not needed for binary stdin?  We use it for binary stdout on windows, so I would expect we'd need something similar for stdin.

What did you find when you were testing this?